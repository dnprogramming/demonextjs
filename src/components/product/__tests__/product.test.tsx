import { render, screen } from '@testing-library/react';
import Product from '../product';
import ProductContent from '@/content/product';
import React from 'react';

describe('Product Component', () => {
    it('renders product header', () => {
        render(<Product />);
        const headerElement = screen.getByText(ProductContent.header);
        expect(headerElement).toBeInTheDocument();
    });

    it('renders product statement', () => {
        render(<Product />);
        const statementElement = screen.getByText(ProductContent.statement);
        expect(statementElement).toBeInTheDocument();
    });

    it('renders product tip tagline', () => {
        render(<Product />);
        const tipTaglineElement = screen.getByText(ProductContent.tipTagline);
        expect(tipTaglineElement).toBeInTheDocument();
    });

    it('renders product list items', () => {
        render(<Product />);
        ProductContent.list.forEach((tip) => {
            const listItemElement = screen.getByText(tip);
            expect(listItemElement).toBeInTheDocument();
        });
    });

    it('renders product component unchanged', () => {
        const { container } = render(<Product />);
        expect(container).toMatchSnapshot();
    });
});