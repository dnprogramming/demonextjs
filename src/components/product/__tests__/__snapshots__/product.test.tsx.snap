// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Product Component renders product component unchanged 1`] = `
<div>
  <div
    id="productHeader"
  >
    <h2>
      Our Products
    </h2>
    <p>
      Products that get made is high quality web applications
    </p>
    <br />
    <h3>
      The Applications
    </h3>
    <p>
      Custom applications always is a very useful item for any business. Having
    </p>
    <br />
    A software craftsman:
    <ol
      type="1"
    >
      <li>
        Always learning new technologies
      </li>
      <li>
        Always honing their craft
      </li>
      <li>
        Takes time to mentor others are they grow
      </li>
    </ol>
  </div>
</div>
`;
